Tests/StarknetTests/Providers/ProviderTests.swift

```diff
@@ -409,9 +411,10 @@ final class ProviderTests: XCTestCase {
     func testBatchGetTransactionByHash() async throws {
         let previousResult = try await provider.getTransactionBy(blockId: .tag(.latest), index: 0).send()
 
-        let transactionsResponse = try await provider.batchRequests(requests:
+        let batchRequest = try provider.batchRequests(requests:
             provider.getTransactionBy(hash: previousResult.transaction.hash!),
-            provider.getTransactionBy(hash: "0x123")).send()
+            provider.getTransactionBy(hash: "0x123"))
+        let transactionsResponse = try await batchRequest.send()
```

Comment:
Might be worth adding a comment here that combining the call to `batchRequests` and `send` into a single statement causes a compiler crash, for future reference to any else wondering why they are separate.

Reply:
Sure, added.